### PR TITLE
Int: More refactoring of internal code

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem

Our config code is complex for its usecases

# This PR's solution

simplify our config code

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: src/orquestra/sdk/_base/_api/_wf_run.py [hunk 1]
```diff
@@ -152,21 +151,9 @@ def start_from_ir(
...
-            raise TypeError(
-                f"'config' argument to `start_from_ir()` has unsupported "
-                f"type {type(config)}."
-            )
-        runtime: RuntimeInterface
-        if _config._runtime_name == "IN_PROCESS":
-            runtime = InProcessRuntime()
-        else:
-            runtime = _config._get_runtime()
+        _config = _resolve_config(config)
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_api/_wf_run.py:154]: **nitpick**: I know this was there before, but could you rename the `_resolve_config` to `resolve_config`? We shouldn't import private functions across modules.

File: src/orquestra/sdk/_base/_config.py [hunk 2]
```diff
@@ -578,37 +463,10 @@ def read_config_names() -> List[str]:
...
-
-    Returns:
-        list: a list of strings, each containing the name of a saved configuration. If
-            the file does not exist, returns an empty list.
-    """
-
-    try:
-        config_file = _open_config_file()
+            config_file = _open_config_file()
+            return [name for name in config_file.configs]
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_config.py:467]: **question**: what's the reason for removing the human-friendly error message on lock timeout?

File: src/orquestra/sdk/_base/_factory.py [hunk 3]
```diff
@@ -28,6 +28,10 @@ class to use.
             project_dir=project_dir,
             config=config,
         )
+    elif config.runtime_name == RuntimeName.IN_PROCESS:
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_factory.py:31]: **praise**: finally!

File: src/orquestra/sdk/_base/_workflow.py [hunk 4]
```diff
@@ -31,6 +31,7 @@
 
 from .. import secrets
 from . import _api, _dsl, loader
+from ._api._config import _resolve_config
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_workflow.py:34]: **nitpick**: we shouldn't import private functions across modules. Please remove the leading `_`.

File: src/orquestra/sdk/_base/_workflow.py [hunk 5]
```diff
@@ -179,20 +178,11 @@ def run(
...
-        else:
-            raise TypeError(
-                f"'config' argument to `run()` has unsupported type {type(config)}."
-            )
-        runtime: RuntimeInterface
-        if _config._runtime_name == "IN_PROCESS":
-            runtime = InProcessRuntime()
-        else:
-            runtime = _config._get_runtime(project_dir=project_dir)
+        _config = _resolve_config(config)
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_workflow.py:181]: **praise**: I've had a hunch this piece of code could be reused!

**Review by alexjuda — CHANGES_REQUESTED** [review]

File: src/orquestra/sdk/_base/_config.py (hunk 2 above)
Comment (SebastianMorawiec) [reply src/orquestra/sdk/_base/_config.py:467]: TLDR is that we use those locks all around the config code, and this is the only place it had this error, which was fine but inconsistent. I dont think it matters that much honestly anyway.

**Review by alexjuda — APPROVED** [review]

**Pull request merged**